include/core/shape.h

```diff
@@ -278,6 +281,15 @@ class Shape : public AutoDerivativeFunction<dim>
   std::string
   point_to_string(const Point<dim> &evaluation_point) const;
 
+  /**
+   * Sets whether the shape is part of a composite
+   */
```

Comment:
missing @param explanation. Missing also @brief. Definition could be clearer here.